internal/provider/kubernetes/gateway.go

```diff
@@ -135,24 +165,30 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 		r.resources.Gateways.Delete(request.NamespacedName)
 	}
 
-	// Set the "Scheduled" condition to true for all accepted gateways.
-	for _, gw := range acceptedGateways {
+	// Set status conditions for all accepted gateways.
+	for i := range acceptedGateways {
+		gw := acceptedGateways[i]
+		// Get the status address of the Gateway's associated Service.
+		svc, err := r.serviceForGateway(ctx, &gw)
```

Comment:
something to consider in the future - another way to solve this would have been for the infra manager to publish a `GatewayStatus` watchable message which would this provider would subscribe to, and update the status when it received a new one

Reply:
@arkodg how would ^ service controller know what Gateway to update status.addresses for?

Reply:
by building the `GatewayStatus` based on the IR metadata info. We might not need a controller as we start off,  just performing a `Get` post a Create/Update might be good enough

Reply:
@arkodg ^ sounds like an interesting approach that requires additional thought and includes unknowns. Since we're close to RC2 and still have a ton of work, I prefer sticking with the current approach. I created https://github.com/envoyproxy/gateway/issues/371 to capture the potential for changing the approach to solving this problem.